ext/tbb/include/tbb.h

```diff
@@ -0,0 +1,302 @@
+/*
+    Copyright (c) 2005-2018 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+
+
+
+*/
+
+/*
+ * Based on: 
+ *
+ *      project: github.com/01org/tbb, files:
+ *      tbb/include/tbb/machine/gcc_generic.h,
+ *      tbb/inclide/tbb/machine/linux-intel64.h
+ *
+ * __TBB mappings:
+ *
+ *      Only added logic that is needed for spin_rw_mutex - only support for
+ *      64b wide data (wordsize == 8) and Linux
+ *
+ *      for Aarch64: default is GCC built-ins (based on gcc version),
+ *      alternative: lockhammer local atomics via USE_LOCAL, with or w/o USE_LSE
+ *
+ *      for x86-64: default is lockhammer local atomics (which should be same
+ *      as machine/linux_intel64.h), aternative: GCC built-ins via
+ *      USE_GCC_BUILTINS (based on gcc version)
+ *
+ * For both ISAs, USE_LOCAL has higher priority over USE_GCC_BUILTINS if used
+ * together.
+ */
+
+
+#ifndef __TBB_H
+#define __TBB_H
+
+#define _GNU_SOURCE
+
+#include "atomics.h"
+
+/* Non default configurations */
+// #define USE_LOCAL
+// #define USE_LSE
+// #define USE_GCC_BUILTINS
+
+
+#ifndef NDEBUG
+#pragma message("Using debug build!!")
+#define DBG(fmt,...) \
```

Comment:
Should we modify this DBG macro a little and make it globally available for all lockhammer workloads? We may need to debug them in the future.

Reply:
That sounds like a good idea but can we do it via a separate issue and new PR rather than folding it into this PR?